src/OpenTelemetry.Extensions.PersistentStorage.Abstractions/PersistentBlobProvider.cs

```diff
@@ -0,0 +1,133 @@
+
+namespace OpenTelemetry.Extensions.PersistentStorage.Abstractions
+{
+    /// <summary>
+    /// Represents persistent blob provider.
+    /// </summary>
+    public abstract class PersistentBlobProvider
+    {
+        /// <summary>
+        /// Attempts to create a new blob with the provided data and lease it.
+        /// </summary>
+        /// <param name="buffer">
+        /// The content to be written.
+        /// </param>
+        /// <param name="leasePeriodMilliseconds">
+        /// The number of milliseconds to lease after the blob is created.
+        /// </param>
+        /// <param name="blob">
+        /// Blob if it is created.
+        /// </param>
+        /// <returns>
+        /// True if the blob was created or else false.
+        /// </returns>
+        public bool TryCreateBlob(byte[] buffer, int leasePeriodMilliseconds, out PersistentBlob blob)
+        {
+            try
+            {
+                return this.OnTryCreateBlob(buffer, leasePeriodMilliseconds, out blob);
+            }
+            catch (Exception)
+            {
+                // TODO: log exception.
+                blob = null;
+                return false;
+            }
+        }
+
+        /// <summary>
+        /// Attempts to create a new blob with the provided data.
+        /// </summary>
+        /// <param name="buffer">
+        /// The content to be written.
+        /// </param>
+        /// <param name="blob">
+        /// Blob if it is created.
+        /// </param>
+        /// <returns>
+        /// True if the blob was created or else false.
+        /// </returns>
+        public bool TryCreateBlob(byte[] buffer, out PersistentBlob blob)
+        {
+            try
+            {
+                return this.OnTryCreateBlob(buffer, out blob);
+            }
+            catch (Exception)
+            {
+                // TODO: log exception;
+                blob = null;
+                return false;
+            }
+        }
+
+        /// <summary>
+        /// Attempts to get a single blob from storage.
+        /// </summary>
+        /// <param name="blob">
+        /// Blob object if found.
+        /// </param>
+        /// <returns>
+        /// True if blob is present or else false.
+        /// </returns>
+        public bool TryGetBlob(out PersistentBlob blob)
+        {
+            try
+            {
+                return this.OnTryGetBlob(out blob);
+            }
+            catch (Exception)
+            {
+                // TODO: log exception.
+                blob = null;
+                return false;
+            }
+        }
+
+        /// <summary>
+        /// Reads a sequence of blobs from storage.
+        /// </summary>
+        /// <returns>
+        /// List of blobs if present in storage or else empty collection.
+        /// </returns>
+        public IEnumerable<PersistentBlob> GetBlobs()
+        {
+            try
+            {
+                return this.OnGetBlobs();
```

Comment:
Could do this here to make sure implementations don't break the contract for returning empty collection:

```csharp
return this.OnGetBlobs() ?? Enumerable.Empty<PersistentBlob>();
```